pkg/metrics/extractfiles.go

```diff
@@ -63,6 +63,6 @@ func (m *FileExtractionMetrics) Emit() {
 		fileExtractionCountMetric.Observe(m.matchCount)
 		fileExtractionMatchCountMetric.Observe(m.fileCount)
 		fileExtractionInaccessibleCountMetric.Observe(m.inaccessibleCount)
+		*m = FileExtractionMetrics{}
```

Comment:
this is the most important one, as without this change, there is potential for nil-dereference